azkaban-common/src/main/java/azkaban/executor/AlerterHolder.java

```diff
@@ -16,119 +16,118 @@
-    final String pluginDir = props.getString("alerter.plugin.dir", "plugins/alerter");
-    allAlerters.putAll(loadPluginAlerters(pluginDir));
-    return allAlerters;
+    // Load additional alerter plugins if any.
+    final String customAlertersBaseDir = props.getString("alerter.plugin.dir", "plugins/alerter");
+    allAlerters.putAll(loadCustomAlerters(customAlertersBaseDir));
+    this.alerters = allAlerters;
+    logger.info("Alerter plugins loaded: {}", this.alerters.keySet());
   }
 
-  private Map<String, Alerter> loadPluginAlerters(final String pluginPath) {
-    final File alerterPluginPath = new File(pluginPath);
-    if (!alerterPluginPath.exists()) {
+  private Map<String, Alerter> loadCustomAlerters(final String alerterPluginsRootPath) {
+    final File pluginsRootPath = new File(alerterPluginsRootPath);
+    if (!pluginsRootPath.exists()) {
       return Collections.<String, Alerter>emptyMap();
     }
 
     final Map<String, Alerter> installedAlerterPlugins = new HashMap<>();
     final ClassLoader parentLoader = getClass().getClassLoader();
-    final File[] pluginDirs = alerterPluginPath.listFiles();
-    final ArrayList<String> jarPaths = new ArrayList<>();
+    final File[] pluginDirs = pluginsRootPath.listFiles();
 
     for (final File pluginDir : pluginDirs) {
-      // load plugin properties
-      final Props pluginProps = PropsUtils.loadPluginProps(pluginDir);
-      if (pluginProps == null) {
-        continue;
-      }
-
-      final String pluginName = pluginProps.getString("alerter.name");
-      final List<String> extLibClassPaths =
-          pluginProps.getStringList("alerter.external.classpaths",
-              (List<String>) null);
-
-      final String pluginClass = pluginProps.getString("alerter.class");
-      if (pluginClass == null) {
-        logger.error("Alerter class is not set.");
-        continue;
-      } else {
-        logger.info("Plugin class " + pluginClass);
+      try {
+        loadCustomAlerter(installedAlerterPlugins, parentLoader, pluginDir);
+      } catch (final Exception e) {
+        logger.error(String.format("Failed to load alerter plugin in '%s'.", pluginDir), e);
       }
+    }
+    return installedAlerterPlugins;
+  }
 
-      Class<?> alerterClass =
-          PluginUtils.getPluginClass(pluginClass, pluginDir, extLibClassPaths, parentLoader);
+  private void loadCustomAlerter(final Map<String, Alerter> installedAlerterPlugins,
+      final ClassLoader parentLoader, final File pluginDir) {
+    // Load plugin properties.
+    final Props pluginProps = PropsUtils.loadPluginProps(pluginDir);
+    if (pluginProps == null) {
+      throw new AzkabanException("Plugin config properties could not be loaded.");
+    }
 
-      if (alerterClass == null) {
-        continue;
-      }
+    final String pluginName = pluginProps.getString("alerter.name", "").trim();
+    if (pluginName.isEmpty()) {
+      throw new AzkabanException("Alerter name is required.");
```

Comment:
anything else in this `pluginProps` can be logged for more info?

Reply:
other interesting properties are logged in the plugin itself, only logging here Azkaban related properties.